pkg/sql/sem/tree/eval.go

```diff
@@ -3130,6 +3130,8 @@ type HasPrivilegeSpecifier struct {
 	// Table privilege
 	TableName *string
 	TableOID  *oid.Oid
+	// Sequences are stored internally as a table
+	IsSequence *bool
```

Comment:
up to you, for a separate PR if needbe

i think i prefer having these all be the zero value instead of nil by default, to avoid the pointer allocation, i.e.
```
type HasPrivilegeSpecifier struct {

	// Database privilege
	DatabaseName string
	DatabaseOID  oid.Oid
	// Table privilege
	TableName string
	TableOID  oid.Oid
	// Sequences are stored internally as a table
	IsSequence bool


  ...
```

i think you'll find if we benchmark it it'll be slightly faster. we have precedence for this elsewhere.

Reply:
Ok, I'll do it separately as part of a post-migration cleanup change